iphone/Classes/TiBuffer.m

```diff
@@ -51,292 +51,292 @@ - (NSArray *)keySequence
-                location:CODELOCATION];
-  }
-
-  void *bytes = [data mutableBytes];
-  for (int i = offsetVal; i < offsetVal + lengthVal; i++) {
-    *(char *)(bytes + i) = byte;
-  }
+    id fillByte = nil;
+    id offset = nil;
+    id length = nil;
+    
+    ENSURE_ARG_AT_INDEX(fillByte, args, 0, NSObject);
+    ENSURE_ARG_OR_NIL_AT_INDEX(offset, args, 1, NSObject);
+    ENSURE_ARG_OR_NIL_AT_INDEX(length, args, 2, NSObject);
+    
+    if (offset != nil && length == nil) {
+        [self throwException:@"TiArgsException"
+                   subreason:@"Bad args to fill(): Expected (int offset, int length), found (int length)"
+                    location:CODELOCATION];
+    }
+    
+    char byte = [TiUtils intValue:fillByte];
+    int offsetVal = [TiUtils intValue:offset];
+    BOOL valid = NO;
+    NSUInteger lengthVal = [TiUtils intValue:length def:0 valid:&valid];
+    if (!valid) {
+        lengthVal = [data length];
+    }
+    
+    if (offsetVal + lengthVal > [data length]) {
+        [self throwException:@"TiBoundsException"
+                   subreason:[NSString stringWithFormat:@"Offset %d plus length %lu extends past bounds (length %lu)", offsetVal, (unsigned long)lengthVal, (unsigned long)[data length]]
+                    location:CODELOCATION];
+    }
+    
+    void *bytes = [data mutableBytes];
+    for (int i = offsetVal; i < offsetVal + lengthVal; i++) {
+        *(char *)(bytes + i) = byte;
+    }
 }
 
 - (void)clear:(id)_void
 {
-  [data resetBytesInRange:NSMakeRange(0, [data length])];
+    [data resetBytesInRange:NSMakeRange(0, [data length])];
 }
 
 - (void)release:(id)_void
 {
-  RELEASE_TO_NIL(data);
+    RELEASE_TO_NIL(data);
 }
 
 - (TiBlob *)toBlob:(id)_void
 {
-  //TODO: Static analysis finds we're leaking the [data copy]. We should have an autorelease here, but for later.
-  return [[[TiBlob alloc] _initWithPageContext:[self executionContext] andData:[[data copy] autorelease] mimetype:@"application/octet-stream"] autorelease];
+    //TODO: Static analysis finds we're leaking the [data copy]. We should have an autorelease here, but for later.
+    return [[[TiBlob alloc] _initWithPageContext:[self executionContext] andData:[[data copy] autorelease] mimetype:@"application/octet-stream"] autorelease];
 }
 
 - (NSString *)toString:(id)_void
 {
-  return [[[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding] autorelease];
+    return [[[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding] autorelease];
 }
 
 #pragma mark Public API : Properties
 
 - (void)setLength:(NSNumber *)length
 {
-  int len = [TiUtils intValue:length];
-  if (len == 0) {
-    RELEASE_TO_NIL(data);
-    return;
-  }
-
-  if (data == nil) {
-    data = [[NSMutableData alloc] initWithLength:len];
-  } else {
-    [data setLength:len];
-  }
+    int len = [TiUtils intValue:length];
```

Comment:
Unrelated to this PR: We should use `NSInteger` and `NSUInteger` instead of `int` in Obj-C.